rocky/rocky/views/upload_yml.py

```diff
@@ -0,0 +1,312 @@
+import yaml
+import io
+from datetime import datetime, timezone
+from typing import Any, ClassVar, TypedDict
+from uuid import uuid4
+
+import yaml.composer
+
+from account.mixins import OrganizationPermissionRequiredMixin, OrganizationView
+from django.contrib import messages
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.urls.base import reverse_lazy
+from django.utils.translation import gettext as _
+from django.views.generic.edit import FormView
+from httpx import HTTPError
+from pydantic import ValidationError
+from tools.forms.upload_yml import YML_ERRORS
+from tools.forms.upload_oois import  UploadOOIYMLForm
+
+from octopoes.api.models import Declaration
+from octopoes.models import OOI, Reference
+from octopoes.models.ooi.config import Config
+from octopoes.models.ooi.dns.records import NXDOMAIN, DNSAAAARecord, DNSARecord, DNSCAARecord, \
+    DNSCNAMERecord, DNSMXRecord, DNSNSRecord, DNSPTRRecord, DNSRecord, DNSSOARecord, DNSTXTRecord
+from octopoes.models.ooi.email_security import DKIMExists, DKIMKey, DKIMSelector, DMARCTXTRecord, \
+    DNSSPFMechanism, DNSSPFMechanismHostname, DNSSPFMechanismIP, DNSSPFMechanismNetBlock, DNSSPFRecord
+from octopoes.models.ooi.findings import FindingType, ADRFindingType, CVEFindingType, CWEFindingType, \
+    CAPECFindingType, RetireJSFindingType, SnykFindingType, KATFindingType, Finding, MutedFinding
+from octopoes.models.ooi.geography import GeographicPoint
+from octopoes.models.ooi.monitoring import Application, Incident
+from octopoes.models.ooi.question import Question
+from octopoes.models.ooi.reports import AssetReport, BaseReport, HydratedReport, Report, ReportData, ReportRecipe
+from octopoes.models.ooi.scans import ExternalScan
+from octopoes.models.ooi.dns.zone import DNSZone, Hostname, ResolvedHostname
+from octopoes.models.ooi.network import AutonomousSystem, IPAddressV4, IPAddressV6, IPV4NetBlock, \
+    IPV6NetBlock, NetBlock, Network, IPAddress, IPPort
+from octopoes.models.ooi.web import URL, HostnameHTTPURL, WebURL, Website, IPAddressHTTPURL, HTTPResource, \
+    HTTPHeader, HTTPHeaderURL, HTTPHeaderHostname, ImageMetadata, RESTAPI, APIDesignRule, APIDesignRuleResult, \
+    SecurityTXT
+from octopoes.models.ooi.service import Service, IPService, TLSCipher
+from octopoes.models.ooi.software import Software, SoftwareInstance
+from octopoes.models.ooi.certificate import SubjectAlternativeName, SubjectAlternativeNameHostname, \
+    SubjectAlternativeNameIP, SubjectAlternativeNameQualifier, X509Certificate
+
+from rocky.bytes_client import get_bytes_client
+
+class OOICandidate(TypedDict):
+    ooi_type: str
+
+class YamlSchape(TypedDict):
+    references: dict[str, dict]
+    oois: list[dict[str, OOICandidate]]
+
+YML_CRITERIA = [
```

Comment:
It looks like it remains based on the previous suggestion of the YAML data shape. I'll change it.

Reply:
It's changed.